evalml/pipelines/components/transformers/feature_selection/feature_selector.py

```diff
@@ -70,3 +70,23 @@ def fit_transform(self, X, y=None):
             pd.DataFrame: Transformed data.
         """
         return self.fit(X, y).transform(X, y)
+
+    def _handle_partial_dependence_fast_mode(self, X, pipeline_parameters):
```

Comment:
if the feature selector decided to drop a feature - should we be calculating partial dependence for it? I'm not sure what the current behavior is but I feel like we shouldn't be if the feature selector deemed the feature unnecessary.  Don't think we need to solve in this PR but might be worth putting a issue up to discuss!

Reply:
I saw your test case! So does this mean features dropped by the feature selector show no impact on predictions? I'm a little confused how the code chunk below affects it.

Reply:
(forgive me if this is now unnecessary after my comment above!)

We can use the information from the original pipeline to determine if the feature gets dropped or not, and if it doesn't get dropped, this code lets us keep it. It basically turns off the Feature Selector for the single column-fitted pipelines. We say "take 100% of features" and "take features above the importance threshold of 0", so that we don't drop the single feature we're fitting that cloned pipeline with! 

It's definitely a bit confusing to look at completely out of context, though. Let me know if you think the docstring isn't descriptive enough.

As to your other point: I don't know that it should be up to the partial dependence function to say that it won't calculate partial dependence for you if it deems the feature to not have an impact on predictions. For all we know, the partial dependence plot could be the way a user discovers this, and I could see it being useful to emit a warning or add something to the plot that indicates why this feature's PD is static. 

If a user wants to not calculate PD at all in this case, I say it's up to them to decide that, but maybe we should be making that information (whether or not a feature gets passed to the estimator at all) more readily available. I can make an issue to think more about the concept of knowing that a feature won't have an impact on predictions and how that information can be accessible to users.

Reply:
no this is an amazing explanation! it might be good to provide a little more context in the docstring 😄

Reply:
Added a `Return` section to the docstring that hopefully explains this a little--let me know if it makes sense to you!

Reply:
Created an issue for adding a feature to show whether a feature will be passed to the estimator: https://github.com/alteryx/evalml/issues/3818